roll/__init__.py

```diff
@@ -132,6 +218,41 @@ def query(self):
             self._query = self.app.Query(parsed_qs)
         return self._query
 
+    def _parse_multipart(self):
+        parser = Multipart(self.app)
+        self._form, self._files = parser.parse(self.content_type)
+        parser.feed_data(self.body)
+
+    def _parse_urlencoded(self):
+        parsed_qs = parse_qs(self.body.decode(), keep_blank_values=True)
+        self._form = self.app.Form(parsed_qs)
+
+    @property
+    def form(self):
+        if self._form is None:
+            if 'multipart/form-data' in self.content_type:
+                self._parse_multipart()
+            elif 'application/x-www-form-urlencoded' in self.content_type:
+                self._parse_urlencoded()
+        return self._form
+
+    @property
+    def files(self):
+        if self._files is None:
+            self._parse_multipart()
+        return self._files
+
+    @property
+    def json(self):
+        try:
+            return json.loads(self.body.decode())
+        except (UnicodeDecodeError, JSONDecodeError):
+            return {}
```

Comment:
Maybe `None` would be a better default?

Reply:
It would raise an error in case of `request.json.get('foo')`.